src/Database/Type/DateTimeType.php

```diff
@@ -304,13 +329,28 @@ public function marshal($value): ?DateTimeInterface
             if ($value === '' || $value === null || is_bool($value)) {
                 return null;
             }
-            $isString = is_string($value);
+
             if (ctype_digit($value)) {
                 return new $class('@' . $value);
-            } elseif ($isString && $this->_useLocaleMarshal) {
-                return $this->_parseLocaleValue($value);
-            } elseif ($isString) {
-                return $this->_parseValue($value);
+            }
+
+            if (is_string($value)) {
+                if ($this->_useLocaleMarshal) {
+                    $dateTime = $this->_parseLocaleValue($value);
+                } else {
+                    $dateTime = $this->_parseValue($value);
+                }
+
+                /** @var \Datetime|\DateTimeImmutable $dateTime */
+                if (
+                    $dateTime !== null &&
+                    $this->localTimezone &&
+                    $this->localTimezone->getName() !== $this->defaultTimezone->getName()
+                ) {
+                    $dateTime = $dateTime->setTimezone($this->defaultTimezone);
```

Comment:
This is not strictly necessary but I think it would be nice to shift the timezone back to the system/default timezone for consistency.

Reply:
Actually it is necessary. Without this one will have to explicitly set `$dbTimezone` too to ensure the timezone if change before converting the value back to string when saving to database, even with PHP and db time are the same.

Reply:
I think we can just set this unconditionally without any cost. The cost is creating the DateTimeZone instance not setting it. I probably wrote code like this elsewhere, but not sure the comparison check is needed.

Reply:
Okay, removing the check would make the code cleaner too.